plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/servlets/ConnectorServletUtil.java

```diff
@@ -0,0 +1,221 @@
+            if (connectorInfo instanceof List<?>) {
+                connectorList.addAll((List<Connector>) connectorInfo);
+            } else if (connectorInfo instanceof ConnectorData) {
+                connectorList.addAll(((ConnectorData) connectorInfo).getConnector());
+            }
+        } catch (KeyManagementException | IOException | NoSuchAlgorithmException e) {
+            log.error("Error while listing connectors : ", e);
+        }
+
+        return new Gson().toJson(connectorList);
+    }
+    
+    /**
+     * Search for the connectors that match the query string
+     * 
+     * @return connector objects
+     */
+    public static String searchConnectors(String queryString) {
+        List<Connector> connectorList = new ArrayList<>();
+        Object connectorInfo;
+        try {
+            connectorInfo = ConnectorStore.searchConnector(getHttpClient(), STORE_URL, queryString);
+            if (connectorInfo instanceof List<?>) {
+                connectorList.addAll((List<Connector>) connectorInfo);
+            } else if (connectorInfo instanceof ConnectorData) {
+                connectorList.addAll(((ConnectorData) connectorInfo).getConnector());
+            }
+        } catch (KeyManagementException | IOException | NoSuchAlgorithmException e) {
+            log.error("Error while listing connectors : ", e);
+        }
+
+        return new Gson().toJson(connectorList);
+    }
+
+    /**
+     * Download the specified connector and update the project
+     * 
+     * @param downloadLink - where to locate the connector
+     * @return true if the connector was successfully downloaded and false if not
+     * 
+     */
+    public static boolean downloadConnectorAndUpdateProjects(String downloadLink) {
+        String zipDestination = null;
+        try {
+            URL url = new URL(downloadLink);
+            String[] segments = downloadLink.split("/");
+            String zipFileName = segments[segments.length - 1];
+            String parentDirectoryPath = ResourcesPlugin.getWorkspace().getRoot().getLocation().toOSString()
+                    + File.separator + DIR_DOT_METADATA + File.separator + DIR_CONNECTORS;
+            File parentDirectory = new File(parentDirectoryPath);
+            if (!parentDirectory.exists()) {
+                parentDirectory.mkdir();
+            }
+            zipDestination = parentDirectoryPath + File.separator + zipFileName;
+            InputStream is = url.openStream();
+            File targetFile = new File(zipDestination);
+            targetFile.createNewFile();
+            FileOutputStream outputStream = new FileOutputStream(targetFile);
+            int bytesRead = -1;
+            byte[] buffer = new byte[BUFFER_SIZE];
+            while ((bytesRead = is.read(buffer)) != -1) {
+                outputStream.write(buffer, 0, bytesRead);
+            }
+            outputStream.close();
```

Comment:
Shouldn't this done in finally block? What will happen if an exception is thrown?

Reply:
Fixed in #1063